lib/DataFetcher/DataFetcher.js

```diff
@@ -0,0 +1,118 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+import { DEFAULT_FETCHER_UPDATE_INTERVAL } from '../utils/constants';
+
+export function DataFetcher({
+  FetcherContext,
+  updateInterval,
+  mutator,
+  resources,
+  resourcesMappingPath,
+  children,
+}) {
+  const mapResourcesToState = ({
+    hasLoaded,
+    isEmpty,
+  }) => {
+    const newContextData = { hasLoaded };
+
+    Object.keys(resources).forEach(key => {
+      newContextData[key] = isEmpty ? [] : get(resources[key], resourcesMappingPath, []);
+    });
+
+    return newContextData;
+  };
+
+  const [contextData, setContextData] = useState(() => mapResourcesToState({
+    hasLoaded: false,
+    isEmpty: true,
+  }));
+
+  const timeoutId = useRef(null);
+
+  const resetFetcher = () => {
+    clearTimeout(timeoutId.current);
+    timeoutId.current = null;
+  };
+
+  const fetchResources = async initial => {
+    const resourcesPromises = Object
+      .values(mutator)
+      .reduce((res, resourceMutator) => res.concat(fetchResource(resourceMutator)), []);
+
+    try {
+      await Promise.all(resourcesPromises);
+
+      setContextData(mapResourcesToState({
+        isEmpty: false,
+        hasLoaded: true,
+      }));
+    } catch (error) {
+      if (initial) {
+        // fill contextData with empty data on unsuccessful initial data retrieval
+        setContextData(mapResourcesToState({
+          isEmpty: true,
+          hasLoaded: true,
+        }));
+      }
+    }
+  };
+
+  useEffect(() => resetFetcher, []);
+
+  useEffect(() => {
+    function updateResourcesData(initial) {
```

Comment:
In some cases (hasLoaded, isEmpty) we use "has" or "is" to specify the boolean variable. But in this case "has" or " is" is not used. I think we should select and use one approach for consistency.

Reply:
Sounds good.